File: class_time_table/src/main/java/xyz/jasenon/lab/class_time_table/dto/DeviceInfoDTO.java
package xyz.jasenon.lab.class_time_table.dto;

import com.alibaba.fastjson2.annotation.JSONField;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 设备信息DTO（解密后的设备信息）
 * 
 * @author Jasenon_ce
 * @date 2026/1/31
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DeviceInfoDTO {
    
    /**
     * 设备ID（唯一标识）
     */
    @JSONField(name = "device_id")
    private String deviceId;
    
    /**
     * 设备名称
     */
    @JSONField(name = "device_name")
    private String deviceName;
    
    /**
     * IP地址（客户端上报，服务端可优先使用 Channel 的 IP）
     */
    @JSONField(name = "ip_address")
    private String ipAddress;

    /**
     * MAC地址
     */
    @JSONField(name = "mac_address")
    private String macAddress;
    
}

File: class-time-table-client/app/src/main/java/xyz/jasenon/classtimetable/network/register/DeviceInfoDTO.kt
package xyz.jasenon.classtimetable.network.register

import com.google.gson.annotations.SerializedName

/**
 * 设备信息 DTO，与后端 RegisterRequest 中 AES 加密内容一致
 */
data class DeviceInfoDTO(
    @SerializedName("device_id")
    val deviceId: String? = null,
    @SerializedName("device_name")
    val deviceName: String? = null,
    @SerializedName("ip_address")
    val ipAddress: String? = null,
    @SerializedName("mac_address")
    val macAddress: String? = null,
    @SerializedName("version")
    val version: String? = null
//    @SerializedName("deviceType")
//    val deviceType: String? = null,
//    @SerializedName("hardwareInfo")
//    val hardwareInfo: String? = null
)
